xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp

```diff
@@ -221,6 +221,18 @@ bool CWebVTTHandler::Initialize()
   return true;
 }
 
+void CWebVTTHandler::Reset()
+{
+  m_previousLines[0].clear();
+  m_previousLines[1].clear();
+  m_previousLines[2].clear();
+  m_currentSection = WebvttSection::UNDEFINED;
+  m_cueCurrentCssStyleSelectors.clear();
```

Comment:
Is it really needed to clear the CSS styles? Looks like some expensive operation to be executed for all decoded packets

Reply:
you're probably right better keep the existing styles, i couldn't find anything specific in the docs about css styles on segmented type
so at this point i think it is also possible that they may never be declared in subsequent segments, but if this happens i will need to check in how handle it better, e.g. overwrite the previous style when it has the same id
i will check

Reply:
Yeah I think this needs further investigation as what you're doing right now is the sledgehammer approach.
Do you have any idea about what properties are kept between each packet decode that actually influence the decoding of the next packet?
I'd say those properties should be cleared **before** decoding the next packet and not as it is now (all properties after decoding each packet)

Reply:
yes clear properties can be done also before
however i had a quick look again and this is what i know:
`m_previousLines` need to be cleared because we parse a new segment file we dont have to look to the previous file data
`m_currentSection` must be set as `UNDEFINED` because the new segment file could have header things that need to be identified and parsed
`m_cueCurrentCssStyleSelectors` my oversight, i see now that can be removed this is temporary already cleaned when the session parsed ends
`m_offset` must be set to 0, a new segment can optionally specify in the header the new timestamp value, if it is not there, it must be zero
for `m_cueCssStyles` i will see next days how could be improve it

the other class members have no influence